catroid/src/test/java/org/catrobat/catroid/test/content/actions/StopRunningStitchActionTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2023 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.content.actions
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.ActionFactory
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.test.MockUtil
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class StopRunningStitchActionTest {
+    var sprite: Sprite? = null
+
+    companion object {
+        private const val FORMULA_TEST_LENGTH = 10
+    }
+
+    @Before
+    fun setUp() {
+        sprite = Sprite("testSprite")
+        val project = Project(MockUtil.mockContextForProject(), "testProject")
+        ProjectManager.getInstance().currentProject = project
+
+        sprite?.actionFactory?.createRunningStitchAction(sprite, SequenceAction(), Formula(FORMULA_TEST_LENGTH))?.act(1.0f)
```

Comment:
Please see if there is a way to start the stitch without the use of the `RunningStitchAction`. Then the test is less coupled to other actions.